hivemind/dht/crypto.py

```diff
@@ -0,0 +1,134 @@
+import base64
+import dataclasses
+import re
+from typing import Optional
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import padding, rsa
+
+from hivemind.utils.serializer import MSGPackSerializer
+
+
+@dataclasses.dataclass(init=True, repr=True, frozen=True)
+class DHTRecord:
+    key: bytes
+    subkey: bytes
```

Comment:
AFAIK, this field is optional

Reply:
It is not optional here, at the protocol layer. In case of sending a non-dictionary value or a whole dictionary, the subkey contains a [special `bytes` value](https://github.com/learning-at-home/hivemind/blob/master/hivemind/dht/protocol.py#L26).